sw/device/tests/pwrmgr_sleep_wake_5_bug_test.c

```diff
@@ -0,0 +1,126 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/ret_sram_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/tests/sim_dv/pwrmgr_sleep_all_wake_ups_impl.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "pwrmgr_regs.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+/*
+  PWRMGR RANDOM SLEEP WAKE UP 5 TEST
+
+  This test runs power manager wake up from normal or deep sleep mode by
+  wake up inputs.
+
+  It deals with wake up 5.
+  5: sensor_ctrl
+
+  For wake up 5 this tests normal and deep sleep in succession.
+
+  This is tracked by a retention sram counter, given there are resets involved.
+ */
+
+OTTF_DEFINE_TEST_CONFIG();
+
+/**
+ * Clean up pwrmgr wakeup reason register for the next round.
+ */
+static void delay_n_clear(uint32_t delay_in_us) {
+  busy_spin_micros(delay_in_us);
+  CHECK_DIF_OK(dif_pwrmgr_wakeup_reason_clear(&pwrmgr));
+}
+
+uint32_t get_wakeup_unit(uint32_t count) { return count / 2; }
```

Comment:
static? same below?

Reply:
or are these used outside this translation unit?

Reply:
Fixed, thanks. Also fixed a few similar cases in related tests.